### ISSUE TITLE
408 request timeout

### ISSUE THREAD

**Issue by davised** [ncbi/datasets#20]
Hi there,

I'm getting a request timeout when running these commands:

```
./datasets-10.1.2 summary genome taxon 'Actinobacteria'
./datasets-10.1.2 summary genome taxon 'Enterobacteriaceae'
```

I'm on the latest version, I renamed the datasets with the version number to keep track.

I have a wrapper to edirect.pl - https://github.com/davised/get_assemblies - and I can do a similar query:

```
get_assemblies organism 'Actinobacteria' --function metadata
```

That one gets 2000 assemblies, and the enterobacteriaceae query gets around 50k assemblies.

Is there a limit (500?) to the query with datasets command? I'd like to move to datasets as a dependency from edirect.pl so my users don't have to have a perl dependency in a python program.

Thanks!

Ed

**Comment by davised** [comment]
```
$ ./datasets-10.1.2 summary genome taxon 'Actinobacteria' --limit none
{"total_count":40150}
```

Hmm so this is significantly more than I was expecting.

In my script, I run an `esearch` command with the taxon name to the genome database, then I get the links to the assembly that way. I suppose that may be restricting the number of genomes that can be otherwise found.

Here's the result from my query with Actinobacteria:

```
<ENTREZ_DIRECT>
  <Db>assembly</Db>
  <WebEnv>MCID_5f921add1687651e75473a78</WebEnv>
  <QueryKey>3</QueryKey>
  <Count>2000</Count>
  <Step>2</Step>
</ENTREZ_DIRECT>
```
Any idea why this might be so different?

Thanks.

**Comment by BradHolmes** [comment]
Hello,

Thank you for giving the datasets CLI a try. I am sorry to hear you ran into some roadblocks. A couple of things I should point out.

First, elink has various limitations, and getting back exactly 2000 requests suggests you have hit a restriction on the number of queries it can process. A better query might be to search the assembly database directly, with something like:

```
esearch -db assembly -query 'Actinobacteria[organism] AND latest[filter]'
```
----
Now, as for using the datasets CLI, I did have success with your summary command:

```
time ./datasets-10.1.2 summary genome taxon 'Actinobacteria' >| Actinobacteria.jsonl
```

That took 15s (albeit, on the internal NCBI network), and resulted in 30MB of json. A quick check shows 40,150 assemblies:

```
cat Actinobacteria.jsonl | jq .assemblies[].assembly.assembly_accession | wc
```

If you are still having timeouts, let me know a datetime of when you executed it, and I'll check our logs.

----

Finally, we are looking for a small community of users to participate in an informal survey and discussion of our tools. Would you or any of your colleagues be interested?

Thanks!

Brad Holmes
Staff Scientist
NCBI/NLM/NIH

**Comment by davised** [comment]
Hi Brad,

Thanks for the insight. I didn't realize the limitation with elink. I retried the same path on the website e.g. search in taxonomy, get links to assembly. It shows the warning that only the first 1000 links can be processed, and it just so happens that produced 2k assemblies.

At any rate, when I tried the same command today, I didn't get the 408 request timeout error. I suppose it was just transient.

I'd be happy to take a survey and forward it to relevant folks here at Oregon State University if it would help you.

I'll rework my script to depend on datasets and see how that goes. I can fix the bug by querying assemblies directly in the meantime.

Thanks again for your quick response! Really saved me some headaches down the road.

Cheers,

Ed
